File: 02-backend/spring-boot-photovoltaic/src/main/java/com/dbwt/springbootphotovoltaic/requestmodels/UpdateProductRequest.java
package com.dbwt.springbootphotovoltaic.requestmodels;

import lombok.Data;

@Data
public class UpdateProductRequest {
    private Long id;

    private String title;

    private String description;

    private Long projectId;

    private Float powerPeak;

    private String orientation;

    private Float inclination;

    private Float area;

    private Float latitude;

    private Float longitude;

    private String img;
}

File: 03-frontend/react-photovoltaic/src/models/UpdateProductRequest.ts
class UpdateProductRequest{
    id: number;
    title: string;
    projectId: number;
    powerPeak: number;
    orientation: string;
    inclination: number;
    area: number;
    longitude: number;
    latitude: number;
    img?: string;
    description: string;

    constructor( id:number, title: string, projectId: number, powerPeak: number, 
        orientation: string, inclination: number, area: number, longitude: number, latitude: number, description:string){

            this.id = id;
            this.title = title;
            this.projectId = projectId;
            this.powerPeak = powerPeak;
            this.orientation = orientation;
            this.inclination = inclination;
            this.area = area;
            this.longitude = longitude;
            this.latitude = latitude;
            this.description = description;
    }

}

export default UpdateProductRequest;